src/build/possible_value.rs

```diff
@@ -148,6 +148,17 @@ impl<'help> PossibleValue<'help> {
         self.help
     }
 
+    /// Get the help specified for this argument, if any and the argument
+    /// value is not hidden
+    #[inline]
+    pub fn get_visible_help(&self) -> Option<&'help str> {
```

Comment:
All of these new functions are specialized for the caller and either should be `pub(crate)` or should live in help generation

Reply:
yes, same is probably true for get visible name (but probably needs to be pub due to usage in `clap_generate`

Reply:
I think you mean `should_show_help`.  Both functions are a bit weird because they couple whether the value is visible with whether the help is visible.  That can make sense in isolated cases but shouldn't be a general API imo.

Reply:
Well `PossibleValue::get_visible_name` would foll into the same categorie, returns `Some(name)` if it is visible and `None` otherwise.

Reply:
yes, it does fall in the same category.  My reference to `should_show_help` was because thats what I saw being used in `clap_complete` unless I was mistaken.

Reply:
get_visible_name as well IIRC

Reply:
Ah, that one already exists which is why I missed it.  The most we can do for that is to update the code and deprecate the existing function.

Reply:
Yeah, I think the idea was to avoid havin this logic at so many places, but as most users will only use clap to builder part of the Api these functions are of little value to them.

 I could just make the methods `pub(crate)` and undo the change in clap-generate.

I wouldn't want to remove them entirely as that would mean implementing their logic all other the different parts of help generation.

Reply:
Sure.

Note that we must keep `get_visible_name` as `pub` to keep compatibility, though we can have `clap_complete` move off of it and deprecate it.

Reply:
yeah, I would just change the ones I added as the other one is already part of the public API.